bundle/bundle.go

```diff
@@ -139,17 +142,28 @@ func ValuesOrDefaults(vals map[string]interface{}, b *Bundle) (map[string]interf
 	res := map[string]interface{}{}
 
 	for name, def := range b.Parameters.Fields {
+		s, ok := b.Definitions[def.Definition]
+		if !ok {
+			return res, fmt.Errorf("unable to find definition for %s", name)
+		}
 		if val, ok := vals[name]; ok {
-			if err := def.ValidateParameterValue(val); err != nil {
-				return res, fmt.Errorf("can't use %v as value of %s: %s", val, name, err)
+			valErrs, err := s.Validate(val)
+			if err != nil {
+				return res, pkgErrors.Wrap(err, fmt.Sprintf("encountered an error validating parameter %s", name))
```

Comment:
nit: If I remember correctly, there's a `pkgErrors .Wrapf` function.

Reply:
You are correct! I'm not sure why I forgot that, thanks for the reminder!